### PR TITLE
LogicButton documentation

### PR DESCRIPTION
**Pull request by ross-pfahler**
Closes RSP-1600

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/button/docs/LogicButton.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## Example
+
+```tsx example
+<LogicButton variant="and">And</LogicButton>
+```
+
+## Content
+
+LogicButtons must have a variant, and optionally can have a label.
```
Comment (LFDanLu) [inline packages/@react-spectrum/button/docs/LogicButton.mdx]: Think you can drop the comma here

File: packages/@react-spectrum/button/docs/LogicButton.mdx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+```tsx example
+<LogicButton variant="or">Test</LogicButton>
+```
+
+### And
+
+```tsx example
+<LogicButton variant="and">Test</LogicButton>
+```
```
Comment (LFDanLu) [inline packages/@react-spectrum/button/docs/LogicButton.mdx]: IMO these examples should be combined under `### Variant`
`### Disabled` is missing here too

File: packages/@react-spectrum/button/docs/LogicButton.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+LogicButtons must have a variant and optionally can have a label.
+A visible label can be provided by passing children.
+
+```tsx example
+<LogicButton variant="or">This or That</LogicButton>
+```
+
+### Accessibility
+
+If no visible label is provided (e.g. an icon only button), an alternative text label must be provided to identify the control for accessibility.
```
Comment (snowystinger) [inline packages/@react-spectrum/button/docs/LogicButton.mdx]: I don't think icons are allowed for LogicButton, so I'm not sure this is needed

[the remark this answers is not in the document]
Comment (ross-pfahler) [reply packages/@react-spectrum/button/docs/LogicButton.mdx]: ok, i have removed this section for now.

File: packages/@react-spectrum/button/docs/LogicButton.mdx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+LogicButtons must have a variant and optionally can have a label.
+A visible label can be provided by passing children.
+
+```tsx example
+<LogicButton variant="or">Or</LogicButton>
+```
+
+### Internationalization
+
+In order to internationalize a LogicButton, a localized string should be passed to the `children` or `aria-label` prop.
```
Comment (ktabors) [inline packages/@react-spectrum/button/docs/LogicButton.mdx]: Do we need a section saying a logic button much have an aria-label if it doesn't have children/label?

Comment (snowystinger) [reply packages/@react-spectrum/button/docs/LogicButton.mdx]: https://github.com/adobe-private/react-spectrum-v3/pull/274#discussion_r393394074

**Pull request merged**